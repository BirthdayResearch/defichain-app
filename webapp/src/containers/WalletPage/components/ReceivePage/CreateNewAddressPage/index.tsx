import React, { FormEvent, useState } from 'react';
import { NavLink } from 'react-router-dom';
import { uid } from 'uid';
import { connect } from 'react-redux';
import { Helmet } from 'react-helmet';
import {
  Button,
  Form,
  FormGroup,
  FormText,
  Label,
  Input,
  Row,
  Col,
  CustomInput,
} from 'reactstrap';
import { I18n } from 'react-redux-i18n';
import { MdArrowBack, MdArrowDropDown, MdArrowDropUp } from 'react-icons/md';
import { WALLET_RECEIVE_PATH } from '../../../../../constants';
import { addReceiveTxnsRequest } from '../../../reducer';
import { getNewAddress } from '../../../service';
import * as log from '../../../../../utils/electronLogger';
import Header from '../../../../HeaderComponent';
import styles from '../../../WalletPage.module.scss';
import {
  getPageTitle,
  isAddressMine,
  isValidAddress,
} from '../../../../../utils/utility';
import { history } from '../../../../../utils/history';
import { PaymentRequestModel } from '@defi_types/rpcConfig';
interface CreateNewAddressPageProps {
  paymentRequests: PaymentRequestModel[];
  addReceiveTxns: (data: any) => void;
}

enum AddressValidity {
  ERROR_EXIST_OWN_WALLET = 'containers.wallet.receivePage.existingAddress',
  ERROR_EXIST_OTHER_WALLET = 'containers.wallet.receivePage.existingAddressOthers',
  ERROR_INVALID_ADDRESS_FORMAT = 'containers.wallet.receivePage.invalidAddress',
}

const CreateNewAddressPage: React.FunctionComponent<CreateNewAddressPageProps> = (
  props: CreateNewAddressPageProps
) => {
  const [label, setLabel] = useState<string>('');
  const [address, setAddress] = useState<string>('');
  const [addressTypeChecked, setAddressTypeChecked] = useState(false);
  const [
    automaticallyGenerateNewAddress,
    setAutomaticallyGenerateNewAddress,
  ] = useState(true);
  const [advanceOpen, setAdvanceOption] = useState(false);
  const [
    addressValidity,
    setAddressValidity,
  ] = useState<AddressValidity | null>();

  const handleChange = (e) => {
    if (e.target.type === 'checkbox') {
      setAutomaticallyGenerateNewAddress(!automaticallyGenerateNewAddress);
      return;
    }
    const { value } = e.target;
    if (value) {
      setLabel(value);
    } else {
      setLabel('');
    }
  };

  const handleChangeAddress = (e) => {
    const { value } = e.target;
    if (value) {
      isAddressValid(value);
      setAddress(value);
    } else {
      setAddress('');
    }
  };

  const isAddressAlreadyExists = (address) => {
    const paymentRequestObj = props.paymentRequests.find(
      (paymentRequest) => paymentRequest.address === address
    );
    return paymentRequestObj != null;
  };

  const isAddressValid = async (value) => {
    try {
      if (value.length < 26 || value.length > 35) {
        setAddressValidity(AddressValidity.ERROR_INVALID_ADDRESS_FORMAT);
        return;
      }
      if (!(await isValidAddress(value))) {
        setAddressValidity(AddressValidity.ERROR_INVALID_ADDRESS_FORMAT);
        return;
      }
      if (isAddressAlreadyExists(value)) {
        setAddressValidity(AddressValidity.ERROR_EXIST_OWN_WALLET);
        return;
      }
      if (!(await isAddressMine(value))) {
        setAddressValidity(AddressValidity.ERROR_EXIST_OTHER_WALLET);
        return;
      }
      setAddressValidity(null);
    } catch (error) {
      log.error(error);
    }
  };

  const onSubmit = async (event: FormEvent) => {
    try {
      event.preventDefault();
      let newAddress;
      if (address) {
        newAddress = address;
      } else {
        newAddress = await getNewAddress(label, addressTypeChecked);
      }
      if (!newAddress) {
        throw new Error(
          I18n.t('containers.wallet.receivePage.addressNotAvailable')
        );
      }
      const data = {
        label,
        id: uid(),
        time: new Date().toString(),
        address: newAddress,
        automaticallyGenerateNewAddress,
        ismine: true,
      };
      props.addReceiveTxns(data);
      history.push(WALLET_RECEIVE_PATH);
    } catch (err) {
      log.error(err);
    }
  };

  return (
    <div className='main-wrapper'>
      <Helmet>
        <title>
          {getPageTitle(
            I18n.t('containers.wallet.receivePage.createNewReceiveAddressLabel')
          )}
        </title>
      </Helmet>
      <Header>
        <Button
          to={WALLET_RECEIVE_PATH}
          tag={NavLink}
          color='link'
          className='header-bar-back'
        >
          <MdArrowBack />
          <span className='d-lg-inline text-uppercase'>
            {I18n.t('containers.wallet.receivePage.backButton')}
          </span>
        </Button>
        {!automaticallyGenerateNewAddress ? (
          <h1>{I18n.t('containers.wallet.receivePage.addReceiveAddress')}</h1>
        ) : (
          <h1>
            {I18n.t(
              'containers.wallet.receivePage.createNewReceiveAddressLabel'
            )}
          </h1>
        )}
      </Header>
      <div className='content'>
        <section>
          <Form onSubmit={onSubmit}>
            <FormGroup check className='mb-5'>
              <Label check className='switch'>
                <Input
                  type='checkbox'
                  name='automaticallyGenerateNewAddress'
                  id='automaticallyGenerateNewAddress'
                  checked={automaticallyGenerateNewAddress}
                  onChange={handleChange}
                />
                &nbsp;
                {I18n.t(
                  'containers.wallet.receivePage.automaticallyGenerateNewAddress'
                )}
              </Label>
            </FormGroup>
            {!automaticallyGenerateNewAddress && (
              <FormGroup className='form-label-group'>
                <Input
                  type='text'
                  value={address}
                  name='address'
                  id='address'
                  onChange={handleChangeAddress}
                  placeholder={I18n.t('containers.wallet.receivePage.address')}
                />
                {address ? (
                  <>
                    {addressValidity == null ? (
                      <FormText color='muted'>
                        {I18n.t('containers.wallet.receivePage.enterAnAddress')}
                      </FormText>
                    ) : (
                      <FormText className={styles['error-dialog']}>
                        {I18n.t(addressValidity)}
                      </FormText>
                    )}
                  </>
                ) : (
                  <FormText color='muted'>
                    {I18n.t('containers.wallet.receivePage.enterAnAddress')}
                  </FormText>
                )}
                <Label for='label'>
                  {I18n.t('containers.wallet.receivePage.address')}
                </Label>
              </FormGroup>
            )}
            <FormGroup className='form-label-group'>
              <Input
                type='text'
                value={label}
                name='label'
                id='label'
                onChange={handleChange}
                placeholder={I18n.t(
                  'containers.wallet.receivePage.addressLabel'
                )}
              />
              <FormText color='muted'>
                {I18n.t('containers.wallet.receivePage.createNewAddressNotice')}
              </FormText>
              <Label for='label'>
                {I18n.t('containers.wallet.receivePage.addressLabel')}
              </Label>
            </FormGroup>
            {automaticallyGenerateNewAddress && (
              <FormGroup
                className='d-flex'
                onClick={() => setAdvanceOption(!advanceOpen)}
              >
                <FormText color='muted'>
                  {I18n.t('containers.wallet.receivePage.showAdvanceOptions')}
                </FormText>
                <Button disabled={true} color='link' size='sm'>
                  {!advanceOpen ? <MdArrowDropDown /> : <MdArrowDropUp />}
                </Button>
              </FormGroup>
            )}
            {advanceOpen && (
              <Row>
                <Col md='4'>
                  <FormGroup>
                    <Label>
                      <strong>
                        {I18n.t('containers.wallet.receivePage.addressType')}
                      </strong>
                    </Label>
                  </FormGroup>
                </Col>
                <Col md='8' lg='6'>
                  <Row className='mb-5'>
                    <Col md='4'>
                      <FormGroup>
                        <CustomInput
                          type='radio'
                          name='addressType'
                          value={'false'}
                          id='addressType1'
                          label={I18n.t(
                            'containers.wallet.receivePage.default'
                          )}
                          checked={!addressTypeChecked}
                          onChange={() => setAddressTypeChecked(false)}
                        />
                      </FormGroup>
                    </Col>
                    <Col md='4'>
                      <FormGroup>
                        <CustomInput
                          type='radio'
                          name='addressType'
                          value={'true'}
                          id='addressType2'
                          label={I18n.t('containers.wallet.receivePage.legacy')}
                          checked={addressTypeChecked}
                          onChange={() => setAddressTypeChecked(true)}
                        />
                      </FormGroup>
                    </Col>
                  </Row>
                </Col>
              </Row>
            )}
          </Form>
        </section>
      </div>
      <footer className='footer-bar'>
        <div className='d-flex justify-content-end'>
          <div>
            <Button
              to={WALLET_RECEIVE_PATH}
              tag={NavLink}
              color='link'
              className='mr-3'
            >
              {I18n.t('containers.wallet.receivePage.cancel')}
            </Button>
            <Button
              color='primary'
              onClick={onSubmit}
              disabled={
                !automaticallyGenerateNewAddress
                  ? addressValidity != null
                  : false
              }
            >
              {I18n.t('containers.wallet.receivePage.createButton')}
            </Button>
          </div>
        </div>
      </footer>
    </div>
  );
};

const mapStateToProps = (state) => {
  const { wallet } = state;

  return {
    paymentRequests: wallet.paymentRequests,
  };
};

const mapDispatchToProps = {
  addReceiveTxns: (data: any) => addReceiveTxnsRequest(data),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateNewAddressPage);
