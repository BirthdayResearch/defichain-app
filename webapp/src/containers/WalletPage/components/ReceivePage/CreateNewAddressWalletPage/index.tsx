import React from 'react';
import uid from 'uid';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { WALLET_RECEIVE_PATH } from '../../../../../constants';
import { addReceiveTxnsRequest } from '../../../reducer';
import { getNewAddress } from '../../../service';
import CreateNewAddress from '../../../../../components/CreateNewAddress';

interface CreateNewAddressPageProps {
  addReceiveTxns: (data: any) => void;
  history: {
    push(url: string): void;
  };
}

const CreateNewAddressWalletPage: React.FunctionComponent<CreateNewAddressPageProps> = (
  props: CreateNewAddressPageProps
) => {
  const onSubmit = async (label: string, addressTypeChecked: boolean) => {
    const newAddress = await getNewAddress(label, addressTypeChecked);
    if (!newAddress) {
      throw new Error(
        I18n.t('containers.wallet.receivePage.addressNotAvailable')
      );
    }
    const data = {
      label,
      id: uid(),
      time: new Date().toString(),
      address: newAddress,
    };
    props.addReceiveTxns(data);
    props.history.push(WALLET_RECEIVE_PATH);
  };

  return (
    <CreateNewAddress
      history={props.history}
      receivePath={WALLET_RECEIVE_PATH}
      handleSubmit={onSubmit}
    />
  );
};

const mapStateToProps = () => {
  return {};
};

const mapDispatchToProps = {
  addReceiveTxns: (data: any) => addReceiveTxnsRequest(data),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateNewAddressWalletPage);
