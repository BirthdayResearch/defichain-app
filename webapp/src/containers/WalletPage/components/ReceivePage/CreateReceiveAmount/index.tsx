import React, { useState } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import uid from 'uid';
import {
  Button,
  Form,
  FormGroup,
  Label,
  Input,
  InputGroup,
  InputGroupAddon,
  InputGroupText,
} from 'reactstrap';
import { MdArrowBack } from 'react-icons/md';
import { NavLink } from 'react-router-dom';
import { I18n } from 'react-redux-i18n';
import * as log from '../../../../../utils/electronLogger';
import { addReceiveTxnsRequest } from '../../../reducer';
import {
  WALLET_PAGE_PATH,
  WALLET_PAYMENT_REQ_BASE_PATH,
  DEFAULT_UNIT,
} from '../../../../../constants';
import { getNewAddress } from '../../../service';
import {
  getAmountInSelectedUnit,
  isLessThanDustAmount,
} from '../../../../../utils/utility';

interface ReceivePageProps {
  history: {
    push(url: string): void;
  };
  unit: string;
  paymentRequests: [];
  addReceiveTxns: (...args: any[]) => void;
}

interface ReceivePageState {
  amount: number | string;
  label: string;
  message: string;
  address: string | undefined;
}

const ReceivePage: React.FunctionComponent<ReceivePageProps> = (
  props: ReceivePageProps
) => {
  const [state, setState] = useState<ReceivePageState>({
    amount: '',
    label: '',
    message: '',
    address: '',
  });

  const setReceivePageState = (updateObj: any) => {
    const updatedState = Object.assign({}, state, updateObj);
    setState(updatedState);
  };

  const onSubmit = async () => {
    try {
      const { amount, label, message } = state;
      const newAddress = await getNewAddress(label);
      if (!newAddress) {
        throw new Error(
          I18n.t('containers.wallet.receivePage.addressNotAvailable')
        );
      }

      const data = {
        label,
        message,
        id: uid(),
        time: new Date().toString(),
        unit: DEFAULT_UNIT,
        address: newAddress,
        amount: amount
          ? getAmountInSelectedUnit(amount, DEFAULT_UNIT, props.unit)
          : null,
      };
      props.addReceiveTxns(data);
      props.history.push(`${WALLET_PAYMENT_REQ_BASE_PATH}/${data.id}`);
    } catch (err) {
      log.error(err);
    }
  };

  const handleChange = (event) => {
    const { name, value, inputMode } = event.target;
    if (inputMode === 'numeric' && isNaN(value) && value !== '') {
      return false;
    }
    const newState = { [name]: value } as Pick<
      ReceivePageState,
      keyof ReceivePageState
    >;
    setReceivePageState(newState);
  };

  const isValidAmount = () => {
    const { amount } = state;
    return amount === '' ? true : !isLessThanDustAmount(amount, props.unit);
  };

  const { amount, label, message } = state;
  return (
    <div className='main-wrapper'>
      <Helmet>
        <title>{I18n.t('containers.wallet.receivePage.receiveDFITitle')}</title>
      </Helmet>
      <header className='header-bar'>
        <Button
          to={WALLET_PAGE_PATH}
          tag={NavLink}
          color='link'
          className='header-bar-back'
        >
          <MdArrowBack />
          <span className='d-lg-inline'>
            {I18n.t('containers.wallet.receivePage.wallet')}
          </span>
        </Button>
        <h1>{I18n.t('containers.wallet.receivePage.receiveDFI')}</h1>
      </header>
      <div className='content'>
        <section>
          <Form>
            <FormGroup className='form-label-group'>
              <InputGroup>
                {/* TODO: show inline error for failed vaildation */}
                <Input
                  type='text'
                  inputMode='numeric'
                  placeholder={I18n.t(
                    'containers.wallet.receivePage.amountToReceive'
                  )}
                  value={amount}
                  name='amount'
                  id='amount'
                  onChange={handleChange}
                  autoFocus
                />
                <Label for='amount'>
                  {!amount
                    ? `${I18n.t(
                        'containers.wallet.receivePage.amount'
                      )} (${I18n.t('containers.wallet.receivePage.optional')})`
                    : I18n.t('containers.wallet.receivePage.amount')}
                </Label>
                <InputGroupAddon addonType='append'>
                  <InputGroupText>{props.unit}</InputGroupText>
                </InputGroupAddon>
              </InputGroup>
            </FormGroup>
            <FormGroup className='form-label-group'>
              <Input
                type='text'
                value={label}
                name='label'
                id='label'
                onChange={handleChange}
                placeholder={I18n.t('containers.wallet.receivePage.label')}
              />
              <Label for='message'>
                {!label
                  ? `${I18n.t('containers.wallet.receivePage.label')} (${I18n.t(
                      'containers.wallet.receivePage.optional'
                    )})`
                  : I18n.t('containers.wallet.receivePage.label')}
              </Label>
            </FormGroup>
            <FormGroup className='form-label-group'>
              <Input
                value={message}
                type='textarea'
                name='message'
                id='message'
                onChange={handleChange}
                placeholder={I18n.t('containers.wallet.receivePage.message')}
                rows='3'
              />
              <Label for='message'>
                {!message
                  ? `${I18n.t(
                      'containers.wallet.receivePage.message'
                    )} (${I18n.t('containers.wallet.receivePage.optional')})`
                  : I18n.t('containers.wallet.receivePage.message')}
              </Label>
            </FormGroup>
          </Form>
        </section>
      </div>
      <footer className='footer-bar'>
        <div className='d-flex justify-content-between align-items-center'>
          <div>
            <div className='caption-secondary'>
              {I18n.t('containers.wallet.receivePage.amountToReceive')}
            </div>
            <div>
              {amount || '-'}&nbsp;
              {props.unit}
            </div>
          </div>
          <div>
            <Button
              to={WALLET_PAGE_PATH}
              tag={NavLink}
              color='link'
              className='mr-3'
            >
              {I18n.t('containers.wallet.receivePage.cancel')}
            </Button>
            <Button
              color='primary'
              onClick={onSubmit}
              disabled={!isValidAmount()}
            >
              {I18n.t('containers.wallet.receivePage.continue')}
            </Button>
          </div>
        </div>
      </footer>
    </div>
  );
};

const mapStateToProps = (state) => {
  const { wallet, settings } = state;
  return {
    paymentRequests: wallet.paymentRequests,
    unit: settings.appConfig.unit,
  };
};

const mapDispatchToProps = {
  addReceiveTxns: (data: any) => addReceiveTxnsRequest(data),
};

export default connect(mapStateToProps, mapDispatchToProps)(ReceivePage);
