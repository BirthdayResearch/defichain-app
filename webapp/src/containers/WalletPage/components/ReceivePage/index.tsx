import React from 'react';
import { Helmet } from 'react-helmet';
import { Button, ButtonGroup } from 'reactstrap';
import { MdArrowBack, MdAdd } from 'react-icons/md';
import PaymentRequestList from './PaymentRequestList';
import { I18n } from 'react-redux-i18n';
import { NavLink as RRNavLink, RouteComponentProps } from 'react-router-dom';
import {
  WALLET_PAGE_PATH,
  WALLET_CREATE_RECEIVE_REQUEST,
} from '../../../../constants';
import Header from '../../../HeaderComponent';
import { getPageTitle } from '../../../../utils/utility';
import { WalletPathEnum } from '../../types';
import { getWalletPathAddress } from '../../service';
import { useDispatch } from 'react-redux';
import { getNewSPVAddress } from '../../reducer';

const ReceivePage: React.FunctionComponent<RouteComponentProps> = (
  props: RouteComponentProps
) => {
  const urlParams = new URLSearchParams(props.location.search);
  const tokenSymbol = urlParams.get(WalletPathEnum.symbol);
  const tokenHash = urlParams.get(WalletPathEnum.hash);
  const tokenAmount = urlParams.get('amount');
  const tokenAddress = urlParams.get(WalletPathEnum.address);
  const isLPS = urlParams.get(WalletPathEnum.isLPS) == 'true';
  const isSPV = urlParams.get(WalletPathEnum.isSPV) == 'true';
  const displayName = urlParams.get(WalletPathEnum.displayName);
  const isDAT = urlParams.get(WalletPathEnum.isDAT) == 'true';

  const dispatch = useDispatch();

  return (
    <div className='main-wrapper'>
      <Helmet>
        <title>
          {getPageTitle(I18n.t('containers.wallet.receivePage.receive'))}
        </title>
      </Helmet>
      <Header>
        <Button
          to={
            tokenSymbol
              ? getWalletPathAddress(
                  WALLET_PAGE_PATH,
                  tokenSymbol,
                  tokenHash || '',
                  tokenAmount || '',
                  tokenAddress || '',
                  isLPS,
                  isSPV,
                  displayName || '',
                  isDAT
                )
              : WALLET_PAGE_PATH
          }
          tag={RRNavLink}
          color='link'
          className='header-bar-back'
        >
          <MdArrowBack />
          <span className='d-lg-inline text-uppercase'>
            {I18n.t('containers.wallet.receivePage.backButton')}
          </span>
        </Button>
        <h1>{I18n.t('containers.wallet.receivePage.receive')}</h1>
        <ButtonGroup>
          {isSPV ? (
            <Button onClick={() => dispatch(getNewSPVAddress())} color='link'>
              <MdAdd />
              <span className='d-lg-inline'>
                {I18n.t('containers.wallet.receivePage.newAddressButton')}
              </span>
            </Button>
          ) : (
            <Button
              to={WALLET_CREATE_RECEIVE_REQUEST}
              tag={RRNavLink}
              color='link'
            >
              <MdAdd />
              <span className='d-lg-inline'>
                {I18n.t('containers.wallet.receivePage.newAddressButton')}
              </span>
            </Button>
          )}
        </ButtonGroup>
      </Header>
      <div className='content'>
        <section>
          <PaymentRequestList isSPV={isSPV} />
        </section>
      </div>
    </div>
  );
};

export default ReceivePage;
