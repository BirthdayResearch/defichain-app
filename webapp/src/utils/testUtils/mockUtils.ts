import PersistentStore from '../persistentStore';
import * as isElectron from '../isElectron';
import * as Notifications from '../notifications';
import log from 'loglevel';
import { runSaga } from 'redux-saga';
import axios from 'axios';

const mockPersistentStore = (getData, setData) => {
  PersistentStore.get = jest.fn(() => getData);
  PersistentStore.set = jest.fn(() => setData);
  return PersistentStore;
};

const mockAxios = mockFunc => {
  axios.create = jest.fn().mockImplementation(() => ({
    post: mockFunc,
  }));
};
const dispatchedFunc = async (method, payload = {}) => {
  const dispatched = Array();
  let action = {};
  if (payload) action = { payload };
  await runSaga(
    {
      dispatch: action => dispatched.push(action),
    },
    method,
    action
  );
  return dispatched;
};

const mockisElectron = () => {
  return jest.spyOn(isElectron, 'isElectron');
};
const mockIpcRenderer = () => {
  return jest.spyOn(isElectron, 'ipcRendererFunc');
};

const spyLogger = () => {
  return jest.spyOn(log, 'error');
};

const mockNotification = () => {
  return (
    jest
      .spyOn(Notifications, 'default')
      // tslint:disable-next-line: no-empty
      .mockImplementation(() => {})
  );
};
export {
  mockPersistentStore,
  mockAxios,
  dispatchedFunc,
  mockisElectron,
  mockIpcRenderer,
  spyLogger,
  mockNotification,
};
