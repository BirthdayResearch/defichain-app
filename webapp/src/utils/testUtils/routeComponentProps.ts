import { createMemoryHistory, createLocation } from 'history';
import { match } from 'react-router';
const history = createMemoryHistory();
const path = `/route/:id`;
const match: match<{ id: string }> = {
  isExact: false,
  path,
  url: path.replace(':id', '1'),
  params: { id: '1' },
};

const location = createLocation(match.url);
delete history.location.key;
delete location.key;

export { history, match, location };
