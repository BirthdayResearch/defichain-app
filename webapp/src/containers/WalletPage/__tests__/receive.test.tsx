import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { history } from '../../../utils/testUtils/routeComponentProps';
import ReceivePage from '../components/ReceivePage';
import { Provider } from 'react-redux';
import store from '../../../app/rootStore';
import { mount } from 'enzyme';

describe('ReceivePage component', () => {
  it('should check for snapshot', () => {
    const wrapper = mount(
      <Router>
        <Provider store={store}>
          <ReceivePage history={history} />
        </Provider>
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
