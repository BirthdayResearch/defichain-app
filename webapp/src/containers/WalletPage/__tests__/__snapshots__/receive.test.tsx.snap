// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceivePage component should check for snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(ReceivePage)
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <ReceivePage
          addReceiveTxns={[Function]}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          paymentRequests={Array []}
          unit="DFI"
        >
          <div
            className="main-wrapper"
          >
            <HelmetWrapper
              defer={true}
              encodeSpecialCharacters={true}
            >
              <SideEffect(NullComponent)
                defer={true}
                encodeSpecialCharacters={true}
                title="Receive DFI – DeFi Blockchain Client"
                titleAttributes={Object {}}
              >
                <NullComponent
                  defer={true}
                  encodeSpecialCharacters={true}
                  title="Receive DFI – DeFi Blockchain Client"
                  titleAttributes={Object {}}
                />
              </SideEffect(NullComponent)>
            </HelmetWrapper>
            <header
              className="header-bar"
            >
              <Button
                className="header-bar-back"
                color="link"
                tag={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "NavLink",
                    "propTypes": Object {
                      "activeClassName": [Function],
                      "activeStyle": [Function],
                      "aria-current": [Function],
                      "className": [Function],
                      "exact": [Function],
                      "innerRef": [Function],
                      "isActive": [Function],
                      "location": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "strict": [Function],
                      "style": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  }
                }
                to="/"
              >
                <NavLink
                  aria-label={null}
                  className="header-bar-back btn btn-link"
                  onClick={[Function]}
                  to="/"
                >
                  <Link
                    aria-current="page"
                    aria-label={null}
                    className="header-bar-back btn btn-link active"
                    onClick={[Function]}
                    style={Object {}}
                    to={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": null,
                      }
                    }
                  >
                    <LinkAnchor
                      aria-current="page"
                      aria-label={null}
                      className="header-bar-back btn btn-link active"
                      href="/"
                      navigate={[Function]}
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <a
                        aria-current="page"
                        aria-label={null}
                        className="header-bar-back btn btn-link active"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        <MdArrowBack>
                          <IconBase
                            attr={
                              Object {
                                "viewBox": "0 0 24 24",
                              }
                            }
                          >
                            <svg
                              fill="currentColor"
                              height="1em"
                              stroke="currentColor"
                              strokeWidth="0"
                              style={
                                Object {
                                  "color": undefined,
                                }
                              }
                              viewBox="0 0 24 24"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                                key="0"
                              />
                            </svg>
                          </IconBase>
                        </MdArrowBack>
                        <span
                          className="d-lg-inline"
                        >
                          Wallet
                        </span>
                      </a>
                    </LinkAnchor>
                  </Link>
                </NavLink>
              </Button>
              <h1>
                Receive DFI
              </h1>
            </header>
            <div
              className="content"
            >
              <section>
                <Form
                  tag="form"
                >
                  <form
                    className=""
                  >
                    <FormGroup
                      className="form-label-group"
                      tag="div"
                    >
                      <div
                        className="form-label-group form-group"
                      >
                        <InputGroup
                          tag="div"
                        >
                          <div
                            className="input-group"
                          >
                            <Input
                              autoFocus={true}
                              id="amount"
                              inputMode="numeric"
                              name="amount"
                              onChange={[Function]}
                              placeholder="Amount to Receive"
                              type="text"
                              value=""
                            >
                              <input
                                autoFocus={true}
                                className="form-control"
                                id="amount"
                                inputMode="numeric"
                                name="amount"
                                onChange={[Function]}
                                placeholder="Amount to Receive"
                                type="text"
                                value=""
                              />
                            </Input>
                            <Label
                              for="amount"
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              <label
                                className=""
                                htmlFor="amount"
                              >
                                Amount (Optional)
                              </label>
                            </Label>
                            <InputGroupAddon
                              addonType="append"
                              tag="div"
                            >
                              <div
                                className="input-group-append"
                              >
                                <InputGroupText
                                  tag="span"
                                >
                                  <span
                                    className="input-group-text"
                                  >
                                    DFI
                                  </span>
                                </InputGroupText>
                              </div>
                            </InputGroupAddon>
                          </div>
                        </InputGroup>
                      </div>
                    </FormGroup>
                    <FormGroup
                      className="form-label-group"
                      tag="div"
                    >
                      <div
                        className="form-label-group form-group"
                      >
                        <Input
                          id="label"
                          name="label"
                          onChange={[Function]}
                          placeholder="Label"
                          type="text"
                          value=""
                        >
                          <input
                            className="form-control"
                            id="label"
                            name="label"
                            onChange={[Function]}
                            placeholder="Label"
                            type="text"
                            value=""
                          />
                        </Input>
                        <Label
                          for="message"
                          tag="label"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <label
                            className=""
                            htmlFor="message"
                          >
                            Label (Optional)
                          </label>
                        </Label>
                      </div>
                    </FormGroup>
                    <FormGroup
                      className="form-label-group"
                      tag="div"
                    >
                      <div
                        className="form-label-group form-group"
                      >
                        <Input
                          id="message"
                          name="message"
                          onChange={[Function]}
                          placeholder="Message"
                          rows="3"
                          type="textarea"
                          value=""
                        >
                          <textarea
                            className="form-control"
                            id="message"
                            name="message"
                            onChange={[Function]}
                            placeholder="Message"
                            rows="3"
                            value=""
                          />
                        </Input>
                        <Label
                          for="message"
                          tag="label"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <label
                            className=""
                            htmlFor="message"
                          >
                            Message (Optional)
                          </label>
                        </Label>
                      </div>
                    </FormGroup>
                  </form>
                </Form>
              </section>
            </div>
            <footer
              className="footer-bar"
            >
              <div
                className="d-flex justify-content-between align-items-center"
              >
                <div>
                  <div
                    className="caption-secondary"
                  >
                    Amount to Receive
                  </div>
                  <div>
                    -
                     
                    DFI
                  </div>
                </div>
                <div>
                  <Button
                    className="mr-3"
                    color="link"
                    tag={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "NavLink",
                        "propTypes": Object {
                          "activeClassName": [Function],
                          "activeStyle": [Function],
                          "aria-current": [Function],
                          "className": [Function],
                          "exact": [Function],
                          "innerRef": [Function],
                          "isActive": [Function],
                          "location": [Function],
                          "onClick": [Function],
                          "replace": [Function],
                          "strict": [Function],
                          "style": [Function],
                          "target": [Function],
                          "to": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    to="/"
                  >
                    <NavLink
                      aria-label={null}
                      className="mr-3 btn btn-link"
                      onClick={[Function]}
                      to="/"
                    >
                      <Link
                        aria-current="page"
                        aria-label={null}
                        className="mr-3 btn btn-link active"
                        onClick={[Function]}
                        style={Object {}}
                        to={
                          Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": null,
                          }
                        }
                      >
                        <LinkAnchor
                          aria-current="page"
                          aria-label={null}
                          className="mr-3 btn btn-link active"
                          href="/"
                          navigate={[Function]}
                          onClick={[Function]}
                          style={Object {}}
                        >
                          <a
                            aria-current="page"
                            aria-label={null}
                            className="mr-3 btn btn-link active"
                            href="/"
                            onClick={[Function]}
                            style={Object {}}
                          >
                            Cancel
                          </a>
                        </LinkAnchor>
                      </Link>
                    </NavLink>
                  </Button>
                  <Button
                    color="primary"
                    disabled={false}
                    onClick={[Function]}
                    tag="button"
                  >
                    <button
                      aria-label={null}
                      className="btn btn-primary"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      Continue
                    </button>
                  </Button>
                </div>
              </div>
            </footer>
          </div>
        </ReceivePage>
      </Connect(ReceivePage)>
    </Provider>
  </Router>
</BrowserRouter>
`;
